packages/flutter/lib/src/material/dialog.dart

```diff
@@ -151,7 +166,11 @@ class Dialog extends StatelessWidget {
             child: Material(
               color: backgroundColor ?? dialogTheme.backgroundColor ?? Theme.of(context).dialogBackgroundColor,
               elevation: elevation ?? dialogTheme.elevation ?? _defaultElevation,
-              shape: shape ?? dialogTheme.shape ?? _defaultDialogShape,
+              shape: shape ?? dialogTheme.shape ?? (
```

Comment:
@darrenaustin I thought I might use a flag via DialogTheme, but the _defaultDialogShape here is actually used as a fallback for the dialogTheme here in the base Dialog class.

I anticipate the turn over being quick, as this is not a severely breaking change. Once this lands I can migrate those affected from the dev roll and remove this flag quickly, the flag itself should not reach stable, instead there will be a migration guide for those that may notice a slight difference in the border radius, or have a golden file test that is affected.

Reply:
Ah, that totally makes sense and I agree that it isn't that much churn.